File: ATM_Python_Sockets_MySQL/ClientProtocol.py

# ERR1: user not found
# ERR2: wrong value
# ERR3: balance less than withdraw price
# ERR4: wrong action
# ERR5: DB error

def processRequest(customer_id, action, price = 0.0):
	outmsg = customer_id + " " + action
	if price > 0.0:
		outmsg += " " + str(price)
	return outmsg

def processReply(inmsg):
	if inmsg == "ERR1":
		return "USER NOT FOUND"
	elif inmsg == "ERR2":
		return "WRONG VALUE"
	elif inmsg == "ERR3":
		return "NOT ENOUGH BALANCE"
	elif inmsg == "ERR4":
		return "WRONG ACTION";
	elif inmsg == "ERR5":
		return "DB ERROR"
	else:
		return inmsg
	

File: ATM_Java_Sockets_MySQL/ClientProtocol.java

public class ClientProtocol {
    /*
    ERR1: user not found
    ERR2: wrong value
    ERR3: balance less than withdraw price
    ERR4: wrong action
    ERR5: DB error
    */

    //for deposit and withdraw request
    public String processRequest(String customer_id, String action, double price){
        String outmsg = customer_id + " " + action + " " + Double.toString(price);
        return outmsg;
    }

    //for balance request
    public String processRequest(String customer_id, String action){
        String outmsg = customer_id + " " + action;
        return outmsg;
    }

    public String processReply(String inmsg){
        switch(inmsg){
            case "ERR1":
                return "USER NOT FOUND";
            case "ERR2":
                return "WRONG VALUE";
            case "ERR3":
                return "NOT ENOUGH BALANCE";
            case "ERR4":
                return "WRONG ACTION";
            case "ERR5":
                return "DB ERROR";
            default:
                return inmsg;
        }
    }
    
}
